webapp/channels/src/components/user_settings/plugin/pluginSetting.tsx

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {get as getPreference} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+
+import SettingItemMax from 'components/setting_item_max';
+import SettingItemMin from 'components/setting_item_min';
+
+import type {GlobalState} from 'types/store';
+
+import RadioInput from './radio';
+import type {PluginConfigurationSetting} from './types';
+
+type Props = {
+    pluginId: string;
+    updateSection: (section: string) => void;
+    activeSection: string;
+    setting: PluginConfigurationSetting;
+}
+
+const PluginSetting = ({
+    pluginId,
+    setting,
+    activeSection,
+    updateSection,
+}: Props) => {
+    const dispatch = useDispatch();
+    const userId = useSelector(getCurrentUserId);
+    const preference = useSelector<GlobalState, string>((state: GlobalState) => getPreference(state, `pp_${pluginId}`, setting.name, setting.default));
+    const [selectedValue, setSelectedValue] = useState(preference);
+
+    const minDescribe = useMemo(() => {
+        if (setting.type === 'radio') {
+            return setting.options.find((v) => v.value === preference)?.text;
+        }
+
+        return undefined;
+    }, [setting, preference]);
+
+    const handleMinUpdateSection = (section: string): void => {
+        updateSection(section);
+    };
+
+    const updateSetting = async (value: string) => {
+        // Save preferences does not offer any await strategy or error handling
+        // so I am leaving this as is for now. We probably should update save
+        // preferences and handle any kind of error or network delay here.
+        dispatch(savePreferences(userId, [{
+            user_id: userId,
+            category: `pp_${pluginId}`.slice(0, 32),
```

Comment:
Not a big fan of the slicing myself. I put some options here: https://community.mattermost.com/private-core/pl/aizg84p6bfykfytz1zc43tedwr

I decided slicing for the time being because it is the most simple approach and "works". But it is too prone to problems.

Reply:
Sounds good to me. Worst case, we can always change something down the line if it actually becomes a problem, but that does seem pretty unlikely the more that I think of it